dbcon/joblist/tuple-bps.cpp

```diff
@@ -2884,8 +2886,26 @@ void TupleBPS::generateJoinResultSet(const vector<vector<Row::Pointer> >& joiner
             {
                 uint32_t dbRoot = outputRG.getDBRoot();
                 uint64_t baseRid = outputRG.getBaseRid();
-                outputData->push_back(rgData);
-                rgData = RGData(outputRG);
+                outputData.push_back(rgData);
```

Comment:
I didn't get why did you decide to move this processing here and why didn't you remove it from TBPS::RMPM?

Reply:
I move it here to get it inside the loop. In the case of a large result set due to many matches, the output buffers can grow to enormous size. By moving everything to inside of generateJoinResultSet, we can release smaller and more manageable rowgroup sizes to the next jobstep.

However, the way generateJoinResultSet is written, it may leave good data in rgDatav that still needs to be processed after return. That same code also processes returned data that is not part of a join and would never enter generateJoinResultSet.

Reply:
I think you will need to reconsider this after I merge Denis'es patch.

Reply:
Maybe